compiler/src/iree/compiler/Dialect/Stream/IR/StreamOpFolders.cpp

```diff
@@ -2275,10 +2275,89 @@ struct ElideNoOpAsyncExecuteOp : public OpRewritePattern<AsyncExecuteOp> {
   }
 };
 
+struct DeduplicateYieldCmdExecuteOp : public OpRewritePattern<AsyncExecuteOp> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AsyncExecuteOp op,
+                                PatternRewriter &rewriter) const override {
+    llvm::SmallVector<bool> keepYield;
+    llvm::SmallVector<Value> yieldOperands;
+    llvm::SmallVector<int> remapping;
+
+    auto yield =
+        cast<IREE::Stream::YieldOp>(op.getBody().front().getTerminator());
+    int64_t oldYieldCount = yield.getResourceOperands().size();
+    for (int i = 0, s = oldYieldCount; i < s; ++i) {
```

Comment:
style/warnings: use consistent types - oldYieldCount is int64_t so you'd use int64_t here - but really, size() is size_t so oldYieldCount and i/s here should be size_t
(some compilers care about this more than others especially if signedness changes, but it also helps readability)